tempesta_fw/ss_skb.c

```diff
@@ -601,19 +681,18 @@ __skb_fragment(struct sk_buff *skb, struct sk_buff *pskb, char *pspt,
  * so use kfree() to release memory.
  */
 int
-ss_skb_get_room(struct sk_buff *skb, char *pspt, unsigned int len, TfwStr *it)
+ss_skb_get_room(SsSkbList *head, struct sk_buff *skb,
+		char *pspt, unsigned int len, TfwStr *it)
 {
-	if (unlikely(len > PAGE_SIZE)) {
-		SS_WARN("Trying to get too large skb room, size=%u\n", len);
-		return -E2BIG;
-	}
+	int r = __skb_fragment(skb, NULL, pspt, len, it);
+	ss_skb_defraglist(head, skb);
```

Comment:
I don't see a problem to pass list head by a function argument. Also you can use some return value.